kore/src/Kore/Reachability/Claim.hs

```diff
@@ -569,17 +570,19 @@ checkImplicationWorker (ClaimPattern.refreshExistentials -> claimPattern) =
     examine ::
         AnyUnified ->
         Pattern RewritingVariableName ->
-        CheckImplicationResult ClaimPattern
+        m (CheckImplicationResult ClaimPattern)
     examine AnyUnified{didAnyUnify} stuck
         | didAnyUnify
           , isBottom condition =
-            Implied
+            pure Implied
         | not didAnyUnify
           , not (isBottom right) =
-            NotImplied claimPattern
-        | otherwise =
-            Lens.set (field @"left") stuck claimPattern
-                & NotImpliedStuck
+            pure $ NotImplied claimPattern
+        | otherwise = do
+            warnClaimRHSIsBottom claimPattern
```

Comment:
We should check if the RHS is bottom before emitting the warning.

Reply:
Obviously, I checked the guards of the preceding cases and falsely concluded that the RHS must always be bottom in the last case. I added the check in 912d7a7